### PR TITLE
XW-1377 | remove Hapi API proxy calls

### PR DESCRIPTION
**Pull request by Warkot**
## Links

* http://wikia-inc.atlassian.net/browse/JIRA-XW1377

## Description

We're getting rid of Hapi and replacing it with Fastboot—that's a step in this direction.
First we need to remove all calls to Hapi API from client side.
In https://github.com/Wikia/mercury/pull/2365 we remove the server side 

## Reviewers
@Wikia/x-wing

### CODE REVIEW

File: front/main/app/models/curated-content.js [hunk 1]
```diff
@@ -25,6 +25,46 @@ import Ember from 'ember';
  * @property {number} height
  */
 
+
```
Comment (tomasznapieralski) [inline front/main/app/models/curated-content.js]: remove extra line

File: front/main/app/models/curated-content.js [hunk 2]
```diff
@@ -25,6 +25,46 @@ import Ember from 'ember';
...
+		});
+	} else if (type === 'category') {
+		const query = {
+			controller: 'ArticlesApi',
+			method: 'getList',
+			expand: 'true',
+			abstract: 0,
+			width: 300,
+			height: 300,
+			category: `${decodeURIComponent(title)}`,
```
Comment (vforge) [inline front/main/app/models/curated-content.js]: Why can't you write it as `category: decodeURIComponent(title)`?

File: front/main/app/components/side-nav-local-wikia-search.js [hunk 3]
```diff
@@ -164,7 +164,14 @@ export default Ember.Component.extend(
 		 * @returns {string}
 		 */
 		getSearchURI(query) {
-			return `${M.prop('apiBase')}/search/${encodeURIComponent(query)}`;
+			return M.buildUrl({
+				path: '/wikia.php',
+				query: {
+					controller: 'MercuryApi',
+					method: 'getSearchSuggestions',
+					query
```
Comment (rogatty) [inline front/main/app/components/side-nav-local-wikia-search.js]: Do we need to use `encodeURIComponent` here or is it irrelevant now?

Comment (bognix) [inline front/main/app/components/side-nav-local-wikia-search.js]: `M.buildUrl` does URL encoding

File: front/main/app/models/curated-content.js (hunk 2 above)
Comment (bognix) [inline front/main/app/models/curated-content.js]: overlooked it - thanks

File: front/main/app/models/curated-content.js [hunk 4]
```diff
@@ -36,28 +76,18 @@ CuratedContentModel.reopenClass({
 	/**
 	 * @param {string} title
 	 * @param {string} [type='section']
-	 * @param {string|null} [offset=null]
 	 * @returns {Ember.RSVP.Promise}
 	 */
-	find(title, type = 'section', offset = null) {
+	find(title, type = 'section') {
```
Comment (rogatty) [inline front/main/app/models/curated-content.js]: We still want to use offset, right?

File: front/main/app/models/curated-content.js [hunk 5]
```diff
@@ -36,28 +76,18 @@ CuratedContentModel.reopenClass({
...
-				params = {};
-
-			let url = `${M.prop('apiBase')}/main/`;
-
-			url += `${type}/${title}`;
-
-			if (offset) {
-				params.offset = offset;
-			}
+				url = getURL(...arguments);
```
Comment (rogatty) [inline front/main/app/models/curated-content.js]: I wouldn't use `...arguments` without good reason. `getURL(title, type, offset)` will work just fine.

File: front/main/tests/unit/models/article-comments-test.js [hunk 6]
```diff
@@ -1,15 +1,77 @@
...
+	beforeEach() {
+		stub = sinon.stub(M, 'buildUrl');
+	},
+
+	afterEach() {
+		M.buildUrl.restore();
+	}
+});
 
 test('url creates valid url to a resource', function (assert) {
```
Comment (rogatty) [inline front/main/tests/unit/models/article-comments-test.js]: I would remove this test completely, IMO it's pointless.

**Comment by rogatty** [comment]
LGTM

File: front/main/tests/unit/models/article-comments-test.js (hunk 6 above)
Comment (bognix) [inline front/main/tests/unit/models/article-comments-test.js]: Let's leave one test case. Better safe than sorry. But the rest of test cases doesn't make any sense

**Comment by hakubo** [comment]
LGTM

**Pull request merged**